File: Examples/DotNET/CSharp/Document/SplitPresentation.cs
using System;
using Com.Aspose.Slides.Api;
using Com.Aspose.Slides.Model;
using Com.Aspose.Storage.Api;

namespace Document
{
    class SplitPresentation
    {
        public static void Run()
        {
            // ExStart:1
            SlidesApi slidesApi = new SlidesApi(Common.APP_KEY, Common.APP_SID, Common.BASEPATH);
            StorageApi storageApi = new StorageApi(Common.APP_KEY, Common.APP_SID, Common.BASEPATH);

            String fileName = "sample-input.pptx";
            int? width = null;
            int? height = null;
            int? from = 2;
            int? to = 3;
            String destFolder = "";
            String format = "png";
            String storage = "";
            String folder = "";

            try
            {
                // Upload source file to aspose cloud storage
                storageApi.PutCreate(fileName, "", "", System.IO.File.ReadAllBytes(Common.GetDataDir() + fileName));

                // Invoke Aspose.Slides Cloud SDK API to split powerpoint presentations
                SplitDocumentResponse apiResponse = slidesApi.PostSlidesSplit(fileName, width, height, to, from, destFolder, format, storage, folder);

                if (apiResponse != null)
                {
                    foreach (ResourceUri slideUri in apiResponse.SplitResult.Slides)
                    {
                        System.Diagnostics.Debug.WriteLine("SlideHref:" + slideUri.Href.Substring(slideUri.Href.LastIndexOf('/') + 1, (slideUri.Href.Length - slideUri.Href.LastIndexOf('/')) - 1));
                        String slidName = slideUri.Href.Substring(slideUri.Href.LastIndexOf('/') + 1, (slideUri.Href.Length - slideUri.Href.LastIndexOf('/')) - 1);
                        // Download document from storage server
                        Com.Aspose.Storage.Model.ResponseMessage storageRes = storageApi.GetDownload(slidName, null, null);
                        System.Diagnostics.Debug.WriteLine("response:" + storageRes.ResponseStream);
                        System.IO.File.WriteAllBytes(Common.GetDataDir() + slidName, storageRes.ResponseStream);
                    }
                    Console.WriteLine("Split PowerPoint Presentations, Done!");
                    Console.ReadKey();
                }
            }
            catch (Exception ex)
            {
                System.Diagnostics.Debug.WriteLine("error:" + ex.Message + "\n" + ex.StackTrace);

            }
            // ExEnd:1
        }
    }
}

File: Examples/Ruby/Document/split_presentation.rb
require 'aspose_slides_cloud'

class Document

  include AsposeSlidesCloud
  include AsposeStorageCloud

  def initialize
    #Get App key and App SID from https://cloud.aspose.com
    AsposeApp.app_key_and_sid("", "")
    @slides_api = SlidesApi.new
  end

  def upload_file(file_name)
    @storage_api = StorageApi.new 
    response = @storage_api.put_create(file_name, File.open("../../../data/" << file_name,"r") { |io| io.read } )
  end

  # Splitting presentations. Create one image per slide.
  def split_presentation
  	file_name = "sample-input.pptx"
  	upload_file(file_name)
  
  	response = @slides_api.post_slides_split(file_name, {to: 3, from: 2, format: "png"})
  end

end

document = Document.new()
puts document.post_slides_split